syft/frameworks/torch/tensors/interpreters/private.py

```diff
@@ -61,6 +60,10 @@ def allow(self, user) -> bool:
         """
         return user in self.allowed_users
 
+    def _before_send(self, *location, user: object = None, **kwargs):
+        if not self.allow(user):
+            raise SendNotPermittedError()
```

Comment:
Re-reading this, it looks like we might want to consolidate `SendNotPermittedError` and `GetNotPermittedError` into one permissions error.

Reply:
After a few reads through, I'm pretty sure this still raises the right exceptions in the right places, and there are tests to check that. Will save any changes to the exceptions for later.